core/src/main/java/org/wso2/testgrid/core/command/RunTestPlanCommand.java

```diff
@@ -314,30 +322,36 @@ private String deriveLogFilePath(Product product, TestPlan testPlan) {
      * @return TestPlan object model
      */
     private TestPlan generateTestPlan(DeploymentPattern deploymentPattern, TestConfig testConfig, String testRepoDir,
-                                      String infraRepoDir) {
-        TestPlan testPlan = new TestPlan();
-        testPlan.setStatus(Status.PENDING);
-        testPlan.setInfraRepoDir(infraRepoDir);
-        testPlan.setTestRepoDir(testRepoDir);
-        testPlan.setDeploymentPattern(deploymentPattern);
-        testPlan.setInfraParameters(testConfig.getInfraParams().get(0).toString());
-        deploymentPattern.addTestPlan(testPlan);
-
-        // Set test run number
-        int latestTestRunNumber = getLatestTestRunNumber(deploymentPattern, testPlan.getInfraParameters());
-        testPlan.setTestRunNumber(latestTestRunNumber + 1);
-
-        // Set test scenarios
-        List<TestScenario> testScenarios = new ArrayList<>();
-        for (String name : testConfig.getScenarios()) {
-            TestScenario testScenario = new TestScenario();
-            testScenario.setName(name);
-            testScenario.setTestPlan(testPlan);
-            testScenario.setStatus(Status.PENDING);
-            testScenarios.add(testScenario);
+                                      String infraRepoDir) throws CommandExecutionException {
+        try {
+            String jsonInfraParams = new ObjectMapper().writeValueAsString(testConfig.getInfraParams().get(0));
```

Comment:
Why do you get the first element here?

Reply:
When the test plan command is run we are only considering a set of infra parameters (only a set of infra params can be there for the test plan). The array which comes here only has 1 element.

Reply:
As I understand, you are saying that testConfig.getInfraParams list always has a single element? Can it be empty?
Can it be more than one? If not, is there a particular reason to have a List as the return type?

Reply:
If this is empty the `jsonInfraParams` will be an empty JSON string. Our input is a YAML file containing a list of infra parameters. We create test plans out of this YAML file. The reason why we are getting a list here is since we have used the same class to read the generated test plans. This way a test plan that is generated out of this process can be fed as an input by the user. These complexities will move away if we ever decide to store the test plans in the DB rather than generating YAML files.

Reply:
This is something we need to fix. But not an issue for this PR.